cache/class-vary-cache.php

```diff
@@ -98,18 +98,26 @@ public static function set_group_for_user( $group, $value ) {
 	/**
 	 * Checks if the request has some in with agroup cookie matching a given group and optionally a value
 	 *
-	 * @param  string $group  Group name.
-	 * @param  string $value Optional - A value for the group.
+	 * @param  string $group Group name.
+	 * @param  string $value (optional) Which segment within the group to check? Omit to check if the user is in the group at all.
 	 *
 	 * @return bool   True on success. False on failure.
 	 */
-	public static function is_user_in_group( $group, $value ) {
+	public static function is_user_in_group( $group, $value = null ) {
 		self::parse_group_cookie();
-		if ( ! isset( self::$groups[ $group ] ) ) {
+
+		// The group isn't defined, or the user isn't in it.
+		if ( empty( self::$groups[ $group ] ) ) {
```

Comment:
Would this break for groups that are named "0"?

Reply:
Yeah, it would, but that's like an unlikely case. We could harden things a bit by changing the default value for groups to `null` and checking explicitly for that?

```suggestion
		if ( ! isset( self::$groups[ $group ] ) || null === self::$groups[ $group ] ) {
```

Although, this maybe gets confusing with the null default for the `$value` param.

Thinking about this some more, maybe we should separate the two uses: `is_user_in_group` (is the user is in the group at all, don't care about the value) vs `is_user_in_group_segment` (is the use in the group with a specific segment value?).

Reply:
Yeah, I think splitting it into two functions would make sense (since we are planning on doing the something similar for `register_group` and `register_groups`

Reply:
@mjangda Can you take a quick peek at the updates I made?

Reply:
Done!